api/v1alpha1/cruisecontroloperation_types.go

```diff
@@ -0,0 +1,80 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/banzaicloud/koperator/api/v1beta1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// CruiseControlOperation is the Schema for the cruiseControlOperation API.
+type CruiseControlOperation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CruiseControlOperationSpec   `json:"spec,omitempty"`
+	Status CruiseControlOperationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+// CruiseControlOperationList contains a list of CruiseControlOperation.
+type CruiseControlOperationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []CruiseControlOperation `json:"items"`
+}
+
+// CruiseControlOperationSpec defines the desired state of CruiseControlOperation.
+type CruiseControlOperationSpec struct {
+	// FailurePolicy defines how failed downscale operations should be handled.
+	// When it is "retry" the koperator re-executes the failed task in every 30 sec.
+	// When it is "ignore" the koperator handles the failed task as completed.
```

Comment:
I think it is better to specify where this default "30 seconds" comes from in this comment

```suggestion
	// When it is "retry", the koperator re-executes the failed task in every 30 seconds (explanation about where this default value comes from).
	// When it is "ignore", the koperator handles the failed task as completed.
```

Reply:
This is documentation, we should not write here where it comes in the codebase.
At first let's try with 30s  I think it is a good default value.

Reply:
Well, this is where the CRD is generated from, right? I don't see why we should not add a short note about where this value comes from. If not here, then where?

Reply:
I wasnt asking for a detailed explanation, I was just thinking about something like `"every 30 seconds (by default)"`

Reply:
Ohh OK. I thought a reference to a place somewhere in the codebase